### ISSUE TITLE
need some code understanding

### ISSUE THREAD

**Issue by helxsz** [lmb-freiburg/mv3d#2]
Hi, after reading the paper, I have some questions

1.  about the label

The paper used  three parameters, azimuth, elevation and distance to express the labels. I assume rad is the distance.   when we use both sin and cos to express the angle, is this way to make the angle unique ?  why can't we express angle as angle = angle / 360 to make it between [0-1] or use sin() or cos() to express the angle ? 

the code is in https://github.com/lmb-freiburg/mv3d/blob/master/utils/tf_utils.py#L159

2. the loss function

https://github.com/lmb-freiburg/mv3d/blob/master/bg_nodm.py#L92

```
        gt_cm = tf.slice(self.images2, [0, 0, 0, 0], [64, 128, 128, 3])
        gt_sm = tf.slice(self.images2, [0, 0, 0, 3], [64, 128, 128, 1])
        sm = gt_sm
        gt_sm = gt_sm * 0.75
        pr_cm = tf.slice(self.gen, [0, 0, 0, 0], [64, 128, 128, 3])
        pr_sm = tf.slice(self.gen, [0, 0, 0, 3], [64, 128, 128, 1])

        self.loss = euclidean_loss(tf.mul(gt_cm, sm), tf.mul(pr_cm, sm)) +\
            0.1 * euclidean_loss(gt_sm, pr_sm)
```

in the first `euclidean_loss(tf.mul(gt_cm, sm), tf.mul(pr_cm, sm))`, gt_cm is the ground truth in color whereas sm is the ground truth in depth, pr_cm is the predication in color , In my understanding it should be like `euclidean_loss(tf.mul(pr_cm, sm), tf.mul(pr_cm, sm))`

in the second `euclidean_loss(gt_sm, pr_sm)`, gt_sm is the  0.75 * ground truth in color, pr_sm is the predicted depth, why these two are used in loss? 

3. about the model

in the convolution layer, the channel are increased from 16,32,64,128,256, my question is why the lower layer has less channels (the lower layer describes more detail feature ) while the higher layers has more channels (the higher layers has more semantic information)

Thanks

**Comment by mtatarchenko** [comment]
Hi,

Answering your questions.

1. Different encodings are possible, your suggestion would likely work as well. One advantage of using sin/cos is that they are continuous cyclic functions, therefore angles 0 and 359 will naturally get similar labels. In case of 'div 360' encoding, the network has to figure this out by itself, which makes the task somewhat more complicated.

2. In this case, the network inputs and the ground truth images contain background. _sm_ is not the ground truth depth, it's the ground truth segmentation mask. It's used to mask out the object background, and thus ignore it during the loss computation. Multiplication by 0.75 simply scales the mask from [0, 1] to [0, 0.75]. We found the latter range to perform better when using _tanh_ after the last convolutional layer. 0.1 is a trade-off between the color map loss, and the segmentation mask loss.

3. It's a very common design choice. Intuitively, as you decrease the feature map resolution from layer to layer in the encoder, you compensate for that by increasing the feature map depth.

Best,
Maxim
